### DISCUSSION TITLE
Is it possible to add a realm level attribute to my access token response?

### DISCUSSION THREAD

**Discussion by anandrhpe** [keycloak/keycloak#28116] [Q&A]
I did a GET request for one of my keycloak realms and got a field called "attributes" in the response. Is it possible for me to add this realm level attribute (called "attributes") to my auth access token response? How do I do it?

**Comment by justin-tay** [comment]
Yes you can add that to your access token. You would need to create a custom OpenID Connect Protocol Mapper or deploy a script, both of which would involve building a JAR and deploying it to the `providers` directory. 

See 
- https://www.baeldung.com/keycloak-custom-protocol-mapper
- https://www.keycloak.org/docs/latest/server_development/#openid-connect-protocol-mapper.

If it's a script it will look something like

```javascript
/**
 * Available variables: 
 * user - the current user
 * realm - the current realm
 * token - the current token
 * userSession - the current userSession
 * keycloakSession - the current keycloakSession
 */

token.setOtherClaims("clientSessionIdleTimeout", realm.getAttribute("clientSessionIdleTimeout"));
token.setOtherClaims("oauth2DeviceCodeLifespan", realm.getAttribute("oauth2DeviceCodeLifespan"));
```

You then configure your custom mapper on the scope that you want.
